### PR TITLE
Fix removing only fully watched videos from playlist

### LINKED ISSUE #7563: Videos can't have "watched" status and/or "Remove watched" doesn't work and/or "playback position" feature is bugged
### Checklist

- [x] I am using the latest version - 0.21.14 
- [x] I checked, but didn't find any duplicates (open OR closed) of this issue in the repo. 
- [x] I have read the contribution guidelines given at https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md.
- [x] This issue contains only one bug (HOPEFULLY). I will open one issue for every bug report I want to file.

### Steps to reproduce the bug
1. Make a playlist and add videos to it
2. Play one or more of those videos till the end or/and use "Mark as watched" from video's context (long tap) menu
3. A red progress bar under each watched video should be visible now (it is not immediately visible in the "What's new" tab so either go to the playlist or press "Refresh", not sure if it isn't another bug)
4. In the playlist view press "Remove watched', then "Yes".

### Actual behavior

- No videos are removed.
- The red progress bar often (not always) misses a chunk from its end, most noticeable on short (3-5 min or less) videos.

### Expected behavior

- All completely watched videos should be removed.
- All completely watched videos should have a full red bar under them.

Just a wild guess - bug could be related to how preloading next video while still playing the current one is implemented, based on those incomplete red bars.

### Device info

 - Android version/Custom ROM version: 5.1.1
 - Device model: SM-J120F

### PR DESCRIPTION
**Pull request by LingYinTianMeng**
#### What is it?
- [x] Bugfix (user facing)
- [ ] Feature (user facing)
- [ ] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR

Before fixing, in the playlist view press "Remove watched', then press "OK", no videos are removed. After fixing, all completely watched videos can be removed.

#### Before/After Screenshots/Screen Record

- Before:

https://user-images.githubusercontent.com/95514488/164049714-6f9da584-5dd1-400e-ba61-277526fc481d.mp4

- After:

https://user-images.githubusercontent.com/95514488/164049736-921ddd74-4249-4a25-b2ab-7141bd03db87.mp4

#### Fixes the following issue(s)

- Fixes #7563 

#### APK testing 

The APK can be found by going to the "Checks" tab below the title. On the left pane, click on "CI", scroll down to "artifacts" and click "app" to download the zip file which contains the debug APK of this PR.

#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by Coder246 — APPROVED** [review]

**Review by Stypox — CHANGES_REQUESTED** [review]
Thank you! I forgot about this when I improved how stream states are saved

File: app/src/main/java/org/schabi/newpipe/local/playlist/LocalPlaylistFragment.java [hunk 1]
```diff
@@ -414,14 +414,21 @@ public void removeWatchedStreams(final boolean removePartiallyWatched) {
...
+                            boolean isFinished = false;
+                            if (streamStateEntity != null) {
+                                isFinished = streamStateEntity.isFinished(duration);
+                            }
+                            final boolean isNotWatchedItem = (streamStateEntity != null
+                                    && !isFinished);
+                            if (indexInHistory < 0) {
+                                notWatchedItems.add(playlistItem);
+                            } else if (isNotWatchedItem) {
                                 notWatchedItems.add(playlistItem);
```
[the remark this answers is not in the document]
Comment (LingYinTianMeng) [reply app/src/main/java/org/schabi/newpipe/local/playlist/LocalPlaylistFragment.java:432]: I think it is    indexInHistory < 0 || (streamStateEntity != null && !streamStateEntity.isFinished(duration)),  there should be 
a "!" before streamStateEntity.isFinished(duration)

[the remark this answers is not in the document]
Comment (Stypox) [reply app/src/main/java/org/schabi/newpipe/local/playlist/LocalPlaylistFragment.java:432]: Ooops, yes, you are right

[the remark this answers is not in the document]
Comment (LingYinTianMeng) [reply app/src/main/java/org/schabi/newpipe/local/playlist/LocalPlaylistFragment.java]: OK, I have modified it.

**Review by Stypox — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/schabi/newpipe/local/playlist/LocalPlaylistFragment.java [hunk 2]
```diff
@@ -414,14 +414,14 @@ public void removeWatchedStreams(final boolean removePartiallyWatched) {
                     } else {
                         final Iterator<StreamStateEntity> streamStatesIter = recordManager
                                 .loadLocalStreamStateBatch(playlist).blockingGet().iterator();
-
                         while (playlistIter.hasNext()) {
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/local/playlist/LocalPlaylistFragment.java:417]:
```suggestion

                        while (playlistIter.hasNext()) {
```

File: app/src/main/java/org/schabi/newpipe/local/playlist/LocalPlaylistFragment.java [hunk 3]
```diff
@@ -414,14 +414,14 @@ public void removeWatchedStreams(final boolean removePartiallyWatched) {
...
                             final PlaylistStreamEntry playlistItem = playlistIter.next();
                             final int indexInHistory = Collections.binarySearch(historyStreamIds,
                                     playlistItem.getStreamId());
-
-                            final boolean hasState = streamStatesIter.next() != null;
-                            if (indexInHistory < 0 || hasState) {
+                            final StreamStateEntity streamStateEntity = streamStatesIter.next();
+                            final long duration = playlistItem.toStreamInfoItem().getDuration();
+                            if (indexInHistory < 0 || (streamStateEntity != null
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/local/playlist/LocalPlaylistFragment.java:423]:
```suggestion

                            if (indexInHistory < 0 || (streamStateEntity != null
```

**Review by Stypox — CHANGES_REQUESTED** [review]
Spaces should be kept. Other than that, thanks :-)

**Review by Stypox — APPROVED** [review]

**Pull request merged**